lib/beaker/command_factory.rb

```diff
@@ -16,7 +16,11 @@ def assertions
     end
 
     def execute(command, options={}, &block)
-      result = self.exec(Command.new(command), options)
+      cmd_create_options = {}
+      exec_opts = options.dup
+      cmd_create_options[:prepend_cmds] = exec_opts.delete(:prepend_cmds) || nil
```

Comment:
This should no longer be an issue because I've updated the code to dup the options hash so that it remains untouched.